source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+            ->execute()
+            ->fetch(\PDO::FETCH_COLUMN);
+
+        return !$hasOrder;
+    }
+}
+```
+First two conditions validates if a customer is logged in and if the customer has already taken an order.
+After the validation passes that the customer is a new customer, the processor collects first all calculated goods in the cart `$goods = $processorCart->getLineItems()->filterGoods();`.
+To calculate the percentage discount for the `new customer discount` the processor uses the shopware core calculator `\Shopware\Bundle\CartBundle\Domain\Price\PercentagePriceCalculator`.
+
+The processor has to be registered over the `cart_processor` container tag. The priority defines at which position the calculator has to be executed (after product calculation, before voucher, ...).
+```
+<service id="cart_extension.new_customer_discount_processor" class="CartExtension\Bundle\CartBundle\Domain\NewCustomerDiscountProcessor">
+    <argument type="service" id="dbal_connection" />
+    <argument type="service" id="shopware_cart.percentage_price_calculator" />
+    <tag name="cart_processor" priority="20000" />
+</service>
+```
+
+### Example - Blacklisted products
+The following examples shows, a possible solution, to prevent that some products can be inside the cart:
+
+```
+<?php
+
+namespace CartExtension\Bundle\CartBundle\Domain;
+
+use Shopware\Bundle\CartBundle\Domain\Cart\Cart;
+use Shopware\Bundle\CartBundle\Domain\Cart\CartContextInterface;
+use Shopware\Bundle\CartBundle\Domain\Cart\CartProcessorInterface;
+use Shopware\Bundle\CartBundle\Domain\Cart\ProcessorCart;
+use Shopware\Bundle\CartBundle\Domain\LineItem\LineItem;
+use Shopware\Bundle\CartBundle\Domain\LineItem\LineItemCollection;
+use Shopware\Bundle\CartBundle\Domain\Product\ProductProcessor;
+
+class BlackListedProductProcessor implements CartProcessorInterface
+{
+    private $blackList = [
+        'SW10239'
+    ];
+
+    public function process(
+        Cart $cart,
+        ProcessorCart $processorCart,
+        CartContextInterface $context
+    ) {
+        /** @var LineItemCollection $products */
+        $products = $cart->getLineItems()->filterType(ProductProcessor::TYPE_PRODUCT);
+
+        /** @var LineItem $product */
+        foreach ($products as $product) {
+            if (in_array($product->getIdentifier(), $this->blackList)) {
+                $cart->getLineItems()->remove($product->getIdentifier());
+            }
+        }
+    }
+}
+```
+
+The service is registered as follow:
+```
+<service id="cart_extension.black_listed_product_processor" class="CartExtension\Bundle\CartBundle\Domain\BlackListedProductProcessor">
+    <tag name="cart_processor" priority="50000" />
+</service>
+```
+Using a high priority defines an early position inside the cart calculation for this processor. The `\Shopware\Bundle\CartBundle\Domain\Product\ProductProcessor` is registered with priority 30000, which means it is executed after this black list processor.
```

Comment:
Using a high priority defines an early position inside the cart calculation for this processor. The `\Shopware\Bundle\CartBundle\Domain\Product\ProductProcessor` is registered with priority 30000, which means it is executed after this blacklist processor.